### PR TITLE
chore: await initialization before routing a msg

### PR DESCRIPTION
**Pull request by bumi**
This uses a deferred promise to make sure the initialization has finished before we process a message call.
This also fixes some LNC issues and makes sure that the WASM is namespaced.

### CODE REVIEW

File: src/extension/background-script/index.ts [hunk 1]
```diff
@@ -154,13 +161,23 @@ async function init() {
...
+init()
+  .then(() => {
+    if (resolveInit) {
+      resolveInit();
+    }
+    if (isFirstInstalled && !state.getState().getAccount()) {
+      utils.openUrl("welcome.html");
+    }
+    if (isRecentlyUpdated) {
+      migrate();
```
Comment (rolznz) [inline src/extension/background-script/index.ts:173]: Should we consider moving migrate() to the bottom of the init() function?

File: src/extension/background-script/index.ts [hunk 2]
```diff
@@ -154,13 +161,23 @@ async function init() {
...
-  if (isFirstInstalled && !state.getState().getAccount()) {
-    utils.openUrl("welcome.html");
-  }
-  if (isRecentlyUpdated) {
-    migrate();
-  }
-});
+init()
+  .then(() => {
+    if (resolveInit) {
```
Comment (rolznz) [inline src/extension/background-script/index.ts:166]: shouldn't resolveInit always exist here? can we throw an exception instead if it undefined?

**Comment by rolznz** [comment]
@bumi the fix works well for me, for a basic test of adding a 10 second sleep to the start of the init function. Without the fix doing this does some strange things, like launching the welcome screen and asking to set a new password.

**Comment by reneaaron** [comment]
Did some testing and added delays to test. Routing calls are stacked during initialization and resolved after that: 👍️

image

File: src/extension/background-script/index.ts (hunk 1 above)
Comment (reneaaron) [reply src/extension/background-script/index.ts:173]: You probably mean to the top of the init function, right? For me it would make sense to await the migration and only then `resolveInit` to avoid calls being executed while migrations are still taking place.

Comment (rolznz) [reply src/extension/background-script/index.ts:173]: I agree `resolveInit` shouldn't be called until the migrations have run. But don't the migrations depend on `init`? that's why I suggested it being moved to the last line in the init() function.

File: src/extension/background-script/index.ts (hunk 2 above)
Comment (reneaaron) [reply src/extension/background-script/index.ts:166]: I think it's just for TS because `resolveInit?`, can't be changed to `resolveInit` because it's only set at runtime I guess.

File: src/extension/background-script/state.ts [hunk 3]
```diff
@@ -96,19 +96,25 @@ const state = createState<State>((set, get) => ({
   },
   getConnector: async () => {
     if (get().connector) {
-      return get().connector as Connector;
+      const connector = (await get().connector) as Connector;
```
Comment (rolznz) [inline src/extension/background-script/state.ts:99]: I think the `if (get().connector)` should be removed and just do this:

````
const connector = (await get().connector) as Connector;
if (connector) {
  return connector;
}
```

Comment (rolznz) [reply src/extension/background-script/state.ts:99]: Ah, I see this won't work for the first request where connector is null.

File: src/app/components/AccountMenu/index.tsx [hunk 4]
```diff
@@ -52,10 +52,10 @@ function AccountMenu({ showOptions = true }: Props) {
   async function selectAccount(accountId: string) {
     setLoading(true);
     try {
-      setAccountId(accountId);
       await msg.request("selectAccount", {
         id: accountId,
       });
+      setAccountId(accountId);
```
Comment (reneaaron) [inline src/app/components/AccountMenu/index.tsx:58]: Why did you change the order here? 

This leads to some side effects (e.g. the avatar changes only after the account has been loaded) :thinking:

Comment (bumi) [reply src/app/components/AccountMenu/index.tsx:58]: is that a problem? 

I wanted to avoid some react dependency to do requests before the new account was switched.
